src/robotFactory.js

```diff
@@ -1,10 +1,98 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    position = {
+      x: 0,
+      y: 0,
+    },
+    chipVersion,
+  ) {
```

Comment:
You will check the presence of properties in the constructor.
You are assigning a new object as a default value, and if you are given an empty object in this argument, this check will not work

```suggestion
  constructor(
    name,
    weight,
    position,
    chipVersion,
  ) {
```